Expose attachments

wetneb opened the pull request:
This is the patch suggested by @noctux in #97. It would be a very useful addition to translation-server, I think.

wetneb committed:
Expose attachments, as suggested by noctux

wetneb commented:
Given the test output, I wonder if a better approach would be to embed the attachments as fields of the record, as originally stored in Zotero?

noctux commented:
Hey, thanks for doing the pullrequest!

On Sun, Mar 24, 2019 at 11:10:21AM -0700, Antonin Delpeuch wrote:
> Given the test output, I wonder if a better approach would be to embed
> the attachments as fields of the record, as originally stored in
> Zotero?

As stated in the original issue, the reason for the current format is
that I tried to represent attachments in a similar fashion to the note
handling in the export. Those are regular item-members as well, but are
represented as dedicated items with a parent relation within the export.
I'm not sure what the primary intention was, but maybe one of the
developers can step in there.

wetneb commented:
@dstillman any preference about the desired output for this? Would you merge this if we update the tests?

noctux commented:
Any updates on this?

noctux committed:
Expose attachment urls via API

Generally, it is not useful to expose regular Zotero attachments via
the API. However, url attachments are an exception: url resources are
of use to the api user, as they help locating various linked online
artifacts that can be readily accessed (in contrast to other attachment
types residing within Zotero's store). Therefore, this commit exports
them as linked items into the API-response.

noctux committed:
Adapt tests to correctly handle url attachments

The previous commit introduced url attachments, which have to be
corretly handled/expected within the tests, which this commit rectifies.

noctux commented:
I've now adapted the tests and fixed one minor flaw in the current implementation (not all url-attachments actually have a title).
@wetneb: Could you pull the current state from my [feature-attachment-urls](https://github.com/noctux/translation-server/tree/feature-attachment-urls) branch and update this pull-request with it?
@dstillman: Would that be enough to get this merged? Or do you see any further problems with this PR?

wetneb committed:
Merge remote-tracking branch 'noctux/feature-attachment-urls' into attachments

noctux commented:
@wetneb Thanks!

dbolser commented:
Any blockers here, or is this just pending a pull? e.g. why hasn't this been pulled?

Thansk

noctux commented:
Just out of curiosity: @dstillman: Any new input on this? Either a definite yes or no? Would it help to make this configurable (with a default to false) to make it more acceptable to publically hosted instances or something like that? Do you want changes to be made to the output format?

dvalde0 commented:
Hi, are there any plans to merge this, or has it already been merged?
